### PR TITLE
feat: expose impression data property

### PR DESCRIPTION
**Pull request by sighphyre**
This exposes the impression data property on the FeatureToggle class through a getter - `hasImpressionData()` this will default to false if not present

### CODE REVIEW

**Comment by sighphyre** [comment]
Hmm, okay, my built in formatter has made changes. Applying the maven task doesn't make changes here, happy to undo the changes by hand if folks object

File: src/test/java/io/getunleash/repository/JsonFeatureToggleParserTest.java [hunk 1]
```diff
@@ -117,6 +117,21 @@ public void should_deserialize_list_of_toggles_with_variants() throws IOExceptio
                 .isNull();
     }
 
+    @Test
+    public void should_should_deserialize_impression_data() throws IOException {
```
Comment (chriswk) [inline src/test/java/io/getunleash/repository/JsonFeatureToggleParserTest.java:121]:
```suggestion
    public void should__deserialize_impression_data() throws IOException {
```

**Review by chriswk — APPROVED** [review]
LGTM

File: src/test/java/io/getunleash/repository/JsonFeatureToggleParserTest.java (hunk 1 above)
Comment (sighphyre) [reply src/test/java/io/getunleash/repository/JsonFeatureToggleParserTest.java]: Yeah, it really really should. It should should

Jokes aside, nice catch! Thanks

**Pull request merged**